CleverFerret/src/main/java/com/universalmedialibrary/ui/books/EnhancedReadingStatisticsViewModel.kt

```
@@ -88,148 +92,87 @@ class EnhancedReadingStatisticsViewModel @Inject constructor(
-            listOf(
-                PublisherCount("Penguin Random House", 15),
-                PublisherCount("HarperCollins", 12),
-                PublisherCount("Simon & Schuster", 10),
-                PublisherCount("Macmillan", 8),
-                PublisherCount("Hachette", 7)
-            )
+            // For now, use sample data - replace with actual database queries
+            metadataDao.getTopPublishers(libraryId).map {
+                PublisherCount(it.publisher, it.count)
+            }
         } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)